packages/server-api/src/coursetypes.ts

```diff
@@ -0,0 +1,41 @@
+import { Position } from '.'
+
+interface DestinationBase {
+  href?: string
+}
+
+export interface Destination extends DestinationBase {
+  position?: Position
+  type?: string
+}
+
+export interface RouteDest extends DestinationBase {
+  reverse?: boolean
+  pointIndex?: number
+  arrivalCircle?: number
+}
+
+export interface ActiveRoute extends DestinationBase {
+  href: string //ActiveRoute always has href
+  pointIndex: number
+  pointTotal: number
+  reverse: boolean
+  name: string
+}
```

Comment:
I looked into the typings here, and I think this is needlessly complex and in fact leads you astray: activateRoute() takes `RouteDest` where href is optional - but that method will throw if href is not there, so the type is wrong.

I think we have just two kinds of Destinations:
- PointDestination (with Position)
- RouteDestination (that is now ActiveRoute, can remain so)

I don't think Destination.type is used anywhere? If we need it the semantics should be clear, needs typing.

Another point is that we are now spreading some of the functionalities and logic over the http route handler and the course api implementation methods. For example clearing the destination happens now in many ways:
- call `destination` with null - this will call clearDestination and emitCourseInfo
- call  `activeRoute` with null - this will call clearDestination and emitCourseInfo
- destination DELETE handler - this will call clearDestination and emitCourseInfo

At the minimum emitCourseInfo should be triggered by `clearDestination`.

I think we should expose `clearDestination` and have `setDestination` and `activateRoute` not accept null. This would also also rhyme better with http api.

These things matter now (types & API structure), as changing the plugin API will be next to impossible or at least messy in the future.

Reply:
> I don't think Destination.type is used anywhere? If we need it the semantics should be clear, needs typing.

Type can be values such as: Waypoint, PoI, Race Mark, etc. as per the spec and the values are not specifically defined anywhere so at this stage they could be anything.

Reply:
> Another point is that we are now spreading some of the functionalities and logic over the http route handler and the course api implementation methods. For example clearing the destination happens now in many ways:
> 
> * call `destination` with null - this will call clearDestination and emitCourseInfo
> * call  `activeRoute` with null - this will call clearDestination and emitCourseInfo
> * destination DELETE handler - this will call clearDestination and emitCourseInfo
> 
> At the minimum emitCourseInfo should be triggered by `clearDestination`.

Committed update to address this.